extensions/CST1229/images.js

```diff
@@ -0,0 +1,382 @@
+(function (Scratch) {
+  "use strict";
+  const QueryImage = {
+    WIDTH: "width",
+    HEIGHT: "height",
+    TOP: "top",
+    BOTTOM: "bottom",
+    LEFT: "left",
+    RIGHT: "right",
+    ROTATION_CENTER_X: "rotation center x",
+    ROTATION_CENTER_Y: "rotation center y",
+  };
+
+  class ImagesExt {
+    constructor(vm) {
+      this.vm = vm;
+      this.render = vm.runtime.renderer;
+      [];
+
+      this.createdImages = new Set();
+      this.validImages = new Set();
+    }
+
+    getInfo() {
+      return {
+        id: "images",
```

Comment:
for any observers, normally such an extension ID would be an issue, but this is port of an existing extension

Reply:
i also completely forgot about that